include/drivers/rtc.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020 Paratronic
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+/**
+ * @file
+ * @brief Public RTC driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <time.h>
+#include <device.h>
+
+typedef int (*rtc_api_get_time_t)(struct device *dev,
+				  struct timespec *tp);
+typedef int (*rtc_api_set_time_t)(struct device *dev,
+				  const struct timespec *tp);
+
+__subsystem struct rtc_driver_api {
+	rtc_api_get_time_t get_time;
+	rtc_api_set_time_t set_time;
+};
+
+/**
+ * @brief Function to get RTC time
+ *
```

Comment:
IMO it's worth giving a little bit more information here about the time.  Specifically I would state that it is applications-specific whether the time maintained is UTC, or offset by a particular zone, or subject to daylight saving adjustments.  Also probably necessary to note that whether time is maintained across sleep or deep-sleep periods.

That's something that should be discussed at the API level.

Reply:
The time represents seconds and nanoseconds since the Epoch. I don't think that localtime must be managed by the RTC driver

Reply:
Which epoch?
* 1970-01-01T00:00:00Z (POSIX)
* 1980-01-05T00:00:00Z (GPS)
* ?? (Bluetooth mesh)

Does the timer advance in UTC seconds? TAI seconds?  They're not the same.

Reply:
For me, epoch is UNIX timestamp like clock_gettime with CLOCK_REALTIME. So, the timer advance in UTC seconds.

Reply:
Advancing in UTC seconds introduces complexity if you have a local RTC that increments based on experienced (TAI) seconds, because you have to adjust based on leap seconds which are announced every 6 months.  The timer is technically monotonic, but does not have a constant slope.  Calculating the duration in experienced seconds between two UTC times is non-trivial.  Supporting it would require functions to detect the number of leap seconds in effect at the time, which depends on a table.  Local time zone is similarly complex, but can derive from UTC given a different table.

For the underlying counter TAI seconds after the POSIX epoch may be the most flexible fixed specification, but the most general is likely to be a description of the time scale given a civil time representation of the epoch (in UTC) and whether the stored time already accounts for subsequent leap seconds or time zone offsets.